From a44d6a165bce59e4e3693c1e62632a28018ef93b Mon Sep 17 00:00:00 2001
From: Alexey Prokhin <alexey@prokhin.ru>
Date: Wed, 17 Oct 2018 19:55:27 +0300
Subject: [PATCH 4/8] winex11.drv: Ignore clip_reset when trying to clip the
 mouse after the desktop has been resized.

This fixes the mouse clipping when the desktop is resized multiple times in a row.

CW-Bug-Id: #21879
---
 dlls/win32u/input.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/win32u/input.c b/dlls/win32u/input.c
index 7c986641968..efc7b6769b4 100644
--- a/dlls/win32u/input.c
+++ b/dlls/win32u/input.c
@@ -2682,7 +2682,7 @@ BOOL clip_fullscreen_window( HWND hwnd, BOOL reset )
 
     if (!ret || !is_fullscreen( &monitor_info, &window_rect )) return FALSE;
     if (is_captured_by_system()) return FALSE;
-    if (NtGetTickCount() - thread_info->clipping_reset < 1000) return FALSE;
+    if (!reset && NtGetTickCount() - thread_info->clipping_reset < 1000) return FALSE;
     if (!reset && clipping_cursor && thread_info->clipping_cursor) return FALSE;  /* already clipping */
 
     if (!grab_fullscreen)
-- 
2.53.0

